src/openms/include/OpenMS/ANALYSIS/XLMS/XFDRAlgorithm.h

```diff
@@ -0,0 +1,246 @@
+
+    /**
+     * @brief Inspects PeptideIdentification pep_id and assigns all cross-link types that this identification belongs to
+     * @param pep_id Peptide ID to be assigned.
+     * @param types Result vector containing the names of the crosslink classes
+     */
+    static void assignTypes_(PeptideHit& ph, StringList& types);
+
+    /** Target counting as performed by the xProphet software package
+     *
+     * @brief xprophet  method for target hits counting as implemented in xProphet
+     * @param cum_histograms Cumulative score distributions
+     */
+    void fdr_xprophet_(std::map< String, Math::Histogram<> >& cum_histograms,
+                      const String& targetclass, const String& decoyclass, const String& fulldecoyclass,
+                      std::vector< double >& fdr, bool mono);
+
+    /**
+    * @brief Calculates the qFDR values for the provided FDR values, assuming that the FDRs are sorted by score in the input vector
+    * @param fdr Vector with FDR values which should be used for qFDR calculation
+    * @param qfdr Result qFDR values
+    */
+    static void calc_qfdr_(const std::vector< double >& fdr, std::vector< double >& qfdr);
+
+    void findTopUniqueHits_(std::vector<PeptideIdentification>& peptide_ids);
+
+    void writeArgumentsLog_() const;
+
+    String getId_(const PeptideHit& ph) const;
+
+    static Size getMinIonsMatched_(const PeptideHit& ph)
+    {
+      Size alpha_ions = Size(ph.getMetaValue("matched_linear_alpha")) + Size(ph.getMetaValue("matched_xlink_alpha"));
+      Size beta_ions = Size(ph.getMetaValue("matched_linear_beta")) + Size(ph.getMetaValue("matched_xlink_beta"));
+      return std::min(alpha_ions, beta_ions);
+    }
+
+    inline static void setIntraProtein_(PeptideHit& ph, const bool value)
+    {
+      ph.setMetaValue("XFDR:is_intraprotein", DataValue(value ? "true" : "false"));
+    }
+
+    inline static void setInterProtein_(PeptideHit& ph, const bool value)
+    {
+      ph.setMetaValue("XFDR:is_interprotein", DataValue(value ? "true" : "false"));
+    }
+
+    /**
+     *  @brief Determines whether the Petide Evidences belong to the same protein, modulo decoy
+     */
+    static bool isSameProtein_(
+            String prot1,
+            String prot2,
+            const String &decoy_string)
+    {
+      prot1.substitute(decoy_string, "");
+      prot2.substitute(decoy_string, "");
+      assert( ! prot1.hasSubstring(decoy_string));
+      assert( ! prot2.hasSubstring(decoy_string));
+      return prot1 == prot2;
+    }
+
+    // Score range for this of the tool
+    Int min_score_;
+    Int max_score_;
+
+    // unique top hits
+    std::vector<String> unique_ids_;
+    std::vector<double> unique_id_scores_;
+
+    // maps index of peptide id all_pep_ids_ to vector of cross link class
+    std::map<String, std::vector<String>> cross_link_classes_;
+
+    // Whether input comes from xQuest
+    bool is_xquest_input_;
```

Comment:
is this needed?  codacy complains about it not being initialized

Reply:
I just noticed that this variable is not used much anymore. I was able to get rid of it.